client/ayon_core/plugins/publish/extract_review.py

```diff
@@ -881,49 +973,183 @@ def ffmpeg_full_args(
 
         return all_args
 
-    def fill_sequence_gaps(self, files, staging_dir, start_frame, end_frame):
-        # type: (list, str, int, int) -> list
+    def fill_sequence_gaps_with_previous(
+        self,
+        collection: str,
+        staging_dir: str,
+        instance: pyblish.plugin.Instance,
+        current_repre_name: str,
+        start_frame: int,
+        end_frame: int
+    ) -> Optional[Dict[int, str]]:
+        """Tries to replace missing frames from ones from last version"""
+        repre_file_paths = self._get_last_version_files(
+            instance, current_repre_name)
+        if repre_file_paths is None:
+            # issues in getting last version files, falling back
+            return None
+
+        prev_collection = clique.assemble(
+            repre_file_paths,
+            patterns=[clique.PATTERNS["frames"]],
+            minimum_items=1
+        )[0][0]
+        prev_col_format = prev_collection.format("{head}{padding}{tail}")
+
+        added_files = {}
+        anatomy = instance.context.data["anatomy"]
+        col_format = collection.format("{head}{padding}{tail}")
+        for frame in range(start_frame, end_frame + 1):
+            if frame in collection.indexes:
+                continue
+            hole_fpath = os.path.join(staging_dir, col_format % frame)
+
+            previous_version_path = prev_col_format % frame
+            # limits too large padding coming from Anatomy
+            previous_version_path = (
+                os.path.join(
+                    anatomy.fill_root(os.path.dirname(previous_version_path)),
```

Comment:
Do we need to fill this root per frame - isn't the root static across the frames so that we need to prepare the folder only once?